src/schema/field_type.rs

```diff
@@ -63,31 +62,23 @@ impl FieldType {
                     }
                 }
             }
-            Json::U64(ref field_val_u64) => {
+            JsonValue::Number(ref field_val_num) => {
                 match *self {
                     FieldType::I64(_) => {
-                        if *field_val_u64 > (i64::max_value() as u64) {
-                            Err(ValueParsingError::OverflowError(format!("Value {:?} is too high for a i64.", field_val_u64)))
+                        if let Some(field_val_i64) = field_val_num.as_i64() {
+                            Ok(Value::I64(field_val_i64))
                         }
                         else {
-                            Ok(Value::I64(*field_val_u64 as i64))
+                            Err(ValueParsingError::OverflowError(format!("Expected an i64 int, got {:?}", json)))
                         }
                     }
                     FieldType::U64(_) => {
-                        Ok(Value::U64(*field_val_u64))
-                    }
-                    _ => {
-                        Err(ValueParsingError::TypeError(format!("Expected a string, got {:?}", json)))
-                    }
-                }
-            },
-            Json::I64(ref field_val_i64) => {
-                match *self {
-                    FieldType::I64(_) => {
-                        Ok(Value::I64(* field_val_i64))
-                    }
-                    FieldType::U64(_) => {
-                        Err(ValueParsingError::TypeError(format!("Expected a positive integer, got {:?}", json)))
+                        if let Some(field_val_u64) = field_val_num.as_u64() {
+                            Ok(Value::U64(field_val_u64))
+                        }
+                        else {
+                            Err(ValueParsingError::TypeError(format!("Expected an u64 int, got {:?}", json)))
```

Comment:
Should these two arms return the same kind of error? Since they'll fail for the same reasons?